components/decimal/src/provider.rs

```diff
@@ -123,6 +112,49 @@ pub struct DecimalSymbolsV1<'data> {
     /// Character used to separate groups in the integer part of the number.
     #[cfg_attr(feature = "serde", serde(borrow))]
     pub grouping_separator: Cow<'data, str>,
+}
+
+// DO NOT SUBMIT THIS IS A TEMPORARY HACK
+// THIS IS SOUND BUT NOT A GREAT USE OF UNSAFE (we should use reusable APIs here)
+#[cfg(feature = "datagen")]
+impl databake::Bake for &DecimalSymbolsV1StringsULE {
+    fn bake(&self, ctx: &databake::CrateEnv) -> databake::TokenStream {
+        use zerovec::ule::VarULE;
+        let buf = self.as_byte_slice();
+        let baked_bytes = buf.bake(ctx);
+        databake::quote!(
+            unsafe { core::mem::transmute::<&[u8], &DecimalSymbolsV1StringsULE>(#baked_bytes) }
+        )
+    }
+}
+
+#[cfg(feature = "datagen")]
+impl databake::BakeSize for &DecimalSymbolsV1StringsULE {
+    fn borrows_size(&self) -> usize {
+        use zerovec::ule::VarULE;
+
+        self.as_byte_slice().borrows_size()
+    }
+}
+
+/// Symbols and metadata required for formatting a [`FixedDecimal`](crate::FixedDecimal).
+///
+/// <div class="stab unstable">
+/// 🚧 This code is considered unstable; it may change at any time, in breaking or non-breaking ways,
+/// including in SemVer minor releases. While the serde representation of data structs is guaranteed
+/// to be stable, their Rust representation might not be. Use with caution.
+/// </div>
+#[icu_provider::data_struct(DecimalSymbolsV1Marker = "decimal/symbols@1")]
+#[derive(Debug, PartialEq, Clone)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize))]
+#[cfg_attr(feature = "datagen", derive(serde::Serialize, databake::Bake))]
+#[cfg_attr(feature = "datagen", databake(path = icu_decimal::provider))]
+pub struct DecimalSymbolsV1<'data> {
```

Comment:
Note: I assume you plan to change this to `DecimalSymbolsV2` yes?

Reply:
We have been consistently renaming things `V2` if the wire format changes, except I guess we failed to do that when it came to the VarZeroVec format change. Still prefer to be explicit about this